mongo/dataset_store.go

```diff
@@ -200,11 +200,24 @@ func (m *Mongo) GetVersions(id, editionID, state string) (*models.VersionResults
 		return nil, errs.ErrVersionNotFound
 	}
 
+	// Remove versions which have not got a state of edition-confirmed,
+	// associated or published
```

Comment:
Can this filter be added to the mongo query?

Reply:
Yes